File: apps/portal-client/src/types/feeSetting.ts
// src/types/feeSetting.ts
export type SettingTypeEnum =
  | 'cashback'
  | 'points'
  | 'loyalty'
  // adicione outros tipos conforme definidos no backend
  ;

/**
 * Base para FeeSetting
 */
export interface FeeSettingBase {
  setting_type: SettingTypeEnum;  // Tipo de serviço
  fee_amount: number | null;      // Valor da taxa (ex: 0.10), null preserva valor atual
}

/**
 * Payload de criação de FeeSetting
 */
export interface FeeSettingCreate extends Omit<FeeSettingBase, 'fee_amount'> {
  fee_amount: number;
}

/**
 * Payload de atualização parcial (upsert)
 */
export interface FeeSettingUpdate {
  fee_amount?: number | null;    // Se omitido, mantém valor atual
}

/**
 * Leitura de FeeSetting
 */
export interface FeeSettingRead extends FeeSettingBase {
  id: string;
  company_id: string;
  created_at: string;  // ISO datetime
  updated_at: string;
}

File: apps/backend/app/schemas/fee_setting.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field
from app.models.fee_setting import SettingTypeEnum
from decimal import Decimal

class FeeSettingBase(BaseModel):
    setting_type: SettingTypeEnum = Field(..., description="Tipo de serviço")
    fee_amount: float           = Field(..., ge=0, description="Valor da taxa (ex: 0.10)")

class FeeSettingCreate(FeeSettingBase):
    pass

class FeeSettingUpdate(BaseModel):
    fee_amount: Decimal | None = Field(None, description="Novo valor da taxa (se omitido, mantém o atual)")

    model_config = ConfigDict()  # from_attributes if estiver lendo do ORM

class FeeSettingRead(FeeSettingBase):
    id: UUID
    company_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
